source/infrastructure/pipeline/e2e-tests/utils/ec2-test-utils.ts

```diff
@@ -0,0 +1,28 @@
+/*****************************************************************************
+ *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.   *
+ *                                                                            *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may   *
+ *  not use this file except in compliance with the License. A copy of the    *
+ *  License is located at                                                     *
+ *                                                                            *
+ *      http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                            *
+ *  or in the 'license' file accompanying this file. This file is distributed *
+ *  on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,        *
+ *  express or implied. See the License for the specific language governing   *
+ *  permissions and limitations under the License.                            *
+ *****************************************************************************/
+import * as ec2 from "@aws-sdk/client-ec2";
+
+export async function getInstanceState(client: ec2.EC2Client, instanceId: string) {
+  const result = await client.send(
+    new ec2.DescribeInstanceStatusCommand({
+      InstanceIds: [instanceId],
+      IncludeAllInstances: true
+    })
+  )
+
+  if(!result.InstanceStatuses) throw new Error(`Instance with id of ${instanceId} Not Found`)
+
+  return result.InstanceStatuses![0].InstanceState!.Name
+}
```

Comment:
Can we get a unit test for this?

Reply:
a unit test for the unit testing util? I suppose we could though I'm not sure it's necessary.

I added the little if catch there to intercept the undefined error that I had originally been having and replacing it with a more useful debugging error should there every be any problems in the future.

Reply:
Yeah, I think we should get into the habit of testing all of our code. Even helpers that are used by tests. One-offs and very simple things might not be necessary (like your wait helpers in another file), but I have a feeling this function will be reused a lot.